x-pack/plugins/apm/public/components/routing/service_detail/index.tsx

```diff
@@ -156,6 +159,12 @@ export const serviceDetail = {
                 offsetRt,
               ]),
             }),
+            locatorPageId: 'serviceTransactionsOverview' as const,
+            defaults: {
+              query: {
+                transactionName: '',
+              },
+            },
```

Comment:
The required query params are filled with default values when they are not present upon page load when a user visits one of the APM UI pages, but `apmRouter.link` requires passing all the required query params or it will fail, in my reasoning to preserve the same semantics when using it, I wanted to either make query params optional in `apmRouter.link` or keep them required and provide default values on each route which is what I went with, let know what do you think!